beater/handlers.go

```diff
@@ -331,15 +361,17 @@ func sendStatus(w http.ResponseWriter, r *http.Request, code int, err error) {
 	w.Header().Set("Content-Type", content_type)
 	w.WriteHeader(code)
 
-	logp.Info("%s	%s %s	%d	%s", extractIP(r), r.Method, r.URL, code, r.Header.Get("User-Agent"))
 	if err == nil {
-		responseValid.Inc()
 		return
 	}
 
-	logp.Err("%s, code=%d", err.Error(), code)
+	logger, ok := r.Context().Value(reqLoggerContextKey).(*logp.Logger)
```

Comment:
@roncohen 
in what case `ok` would be false?
what is the benefit of putting and getting the logger in the context instead of doing another `logger := logp.NewLogger("request")` here? doesn't seem an expensive operation

Reply:
`ok` would only be false if `logHandler` somehow wasn't applied. I didn't want to run the risk of breaking a customer app in production, so i did this fallback here. We can do a `logp.NewLogger("request").Err(..)` here if `ok` is false. 

I put the logger in the context because the logger has context that we don't otherwise pass around for example `request_id`.

Reply:
got it, thanks